src/main/java/io/confluent/connect/hdfs/hive/HiveMetaStore.java

```diff
@@ -71,19 +71,10 @@ public HiveMetaStore(Configuration conf, HdfsSinkConnectorConfig connectorConfig
     R call() throws TException;
   }
 
-  private <R> R doWithRetry(ClientAction<R> action) throws TException {
-    try {
-      return action.call();
-    } catch (TException e) {
-      try {
-          client.reconnect();
-      } catch (MetaException swallowedException) {
-        // reconnect failed, throw the original exception
-        log.warn("Cannot connect to Hive meta store: {}", swallowedException.getMessage());
-        throw e;
-      }
-      return action.call();
-    }
+  private <R> R doAction(ClientAction<R> action) throws TException {
+    // No need to implement retries. We use RetryingHiveMetaStoreClient
+    // which already implements the retrying and reconnect logic
```

Comment:
That makes sense. I think it worthwhile to remove the retry in the HDFS connector's `HiveMetaStore`.  Would you like to update the comments to elaborate how the `RetryingHiveMetaStoreClient` handles retry?